From: Jeff Epler <jepler@unpythonic.net>
Date: Fri, 19 Apr 2013 08:15:18 -0500
Subject: Create kfreebsd like linux in most respects

---
 SConstruct | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/SConstruct
+++ b/SConstruct
@@ -38,7 +38,7 @@
 # command-line. Treat this output as the value of HOST_OS
 def get_running_os_name():
     running_os = os.sys.platform
-    if running_os.startswith('linux'):
+    if running_os.startswith('linux') or running_os.startswith('gnukfreebsd'):
         running_os = 'linux'
     elif running_os.startswith('freebsd'):
         running_os = 'freebsd'
