From: Jeff Epler <jepler@unpythonic.net>
Date: Fri, 19 Apr 2013 08:15:18 -0500
Subject: Create kfreebsd like linux in most respects

---
 SConstruct | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 5fb0e83..f3ac3f9 100644
--- a/SConstruct
+++ b/SConstruct
@@ -344,6 +344,8 @@ libdeps.setup_environment( env )
 
 if env['PYSYSPLATFORM'] == 'linux3':
     env['PYSYSPLATFORM'] = 'linux2'
+if 'gnukfreebsd' in env['PYSYSPLATFORM']:
+    env['PYSYSPLATFORM'] = 'linux2'
 if 'freebsd' in env['PYSYSPLATFORM']:
     env['PYSYSPLATFORM'] = 'freebsd'
 
@@ -524,7 +526,7 @@ if "darwin" == os.sys.platform:
        env.Append( EXTRACPPPATH=filterExists(["/sw/include" , "/opt/local/include"]) )
        env.Append( EXTRALIBPATH=filterExists(["/sw/lib/", "/opt/local/lib"]) )
 
-elif os.sys.platform.startswith("linux"):
+elif os.sys.platform.startswith("linux") or os.sys.platform.startswith("gnukfreebsd"):
     linux = True
     platform = "linux"
 
