From: Antonin Kral <a.kral@bobek.cz>
Date: Wed, 14 Sep 2011 17:26:58 +0200
Subject: Ignore unused but set variables and params (Closes: #625401)

---
 SConstruct |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 5c48996..9d52617 100644
--- a/SConstruct
+++ b/SConstruct
@@ -704,7 +704,8 @@ if nix:
     # env.Append( " -Wconversion" ) TODO: this doesn't really work yet
     if linux:
         env.Append( CPPFLAGS=" -Werror " )
-        if not has_option('clang'): 
+        env.Append( CPPFLAGS=" -Wno-error=unused-but-set-variable -Wno-error=unused-but-set-parameter " ) # ignore new gcc 4.6 warnings
+        if not has_option('clang'):
             env.Append( CPPFLAGS=" -fno-builtin-memcmp " ) # glibc's memcmp is faster than gcc's
 
     env.Append( CPPDEFINES="_FILE_OFFSET_BITS=64" )
