From: Antonin Kral <a.kral@bobek.cz>
Date: Tue, 5 Apr 2011 10:19:55 +0200
Subject: return empty TERM if not set (Closes: #620910)

---
 SConstruct |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/SConstruct b/SConstruct
index e142f29..b5a2e30 100644
--- a/SConstruct
+++ b/SConstruct
@@ -714,7 +714,7 @@ if nix:
 
     #make scons colorgcc friendly
     env['ENV']['HOME'] = os.environ['HOME']
-    env['ENV']['TERM'] = os.environ['TERM']
+    env['ENV']['TERM'] = os.environ.get('TERM', '')
 
     if linux and has_option( "sharedclient" ):
         env.Append( LINKFLAGS=" -Wl,--as-needed -Wl,-zdefs " )
