From: Antonin Kral <a.kral@bobek.cz>
Date: Sun, 7 Aug 2011 19:19:38 +0200
Subject: (Closes: #625401) ignore unused-but-set-{variable,parameter}

---
 SConstruct |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/SConstruct b/SConstruct
index 0293366..bafdac4 100644
--- a/SConstruct
+++ b/SConstruct
@@ -673,6 +673,7 @@ if nix:
     if linux:
         env.Append( CPPFLAGS=" -Werror " )
         env.Append( CPPFLAGS=" -fno-builtin-memcmp " ) # glibc's memcmp is faster than gcc's
+        env.Append( CPPFLAGS=" -Wno-error=unused-but-set-variable -Wno-error=unused-but-set-parameter " ) # ignore new gcc 4.6
     env.Append( CXXFLAGS=" -Wnon-virtual-dtor " )
     env.Append( LINKFLAGS=" -fPIC -pthread -rdynamic" )
     env.Append( LIBS=[] )
-- 
