From: Jeff Epler <jepler@unpythonic.net>
Date: Fri, 19 Apr 2013 08:16:31 -0500
Subject: All platforms but Windows find hash in std::tr1

---
 src/third_party/s2/hash.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/third_party/s2/hash.h b/src/third_party/s2/hash.h
index a255f2c..d399bb8 100644
--- a/src/third_party/s2/hash.h
+++ b/src/third_party/s2/hash.h
@@ -7,10 +7,10 @@
 #include "mongo/platform/unordered_set.h"
 #define hash_set mongo::unordered_set
 
-#if defined OS_LINUX || defined OS_MACOSX || defined __sunos__ || defined __freebsd__
+#if ! defined OS_WINDOWS
 #define HASH_NAMESPACE_START namespace std { namespace tr1 {
 #define HASH_NAMESPACE_END }}
-#elif defined OS_WINDOWS
+#else
 #define HASH_NAMESPACE_START namespace std {
 #define HASH_NAMESPACE_END }
 #endif
