From: Antonin Kral <a.kral@bobek.cz>
Date: Thu, 15 Dec 2011 11:22:43 +0100
Subject: use system-wide pcre

---
 SConstruct |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/SConstruct b/SConstruct
index fcd9c70..937bf72 100644
--- a/SConstruct
+++ b/SConstruct
@@ -37,7 +37,11 @@ def getThirdPartyShortNames():
     for x in os.listdir( "third_party" ):
         if not x.endswith( ".py" ) or x.find( "#" ) >= 0:
             continue
-         
+
+        #use system-wide pcre
+        if x.find('pcre') >= 0:
+            continue
+
         lst.append( x.rpartition( "." )[0] )
     return lst
 
@@ -511,6 +515,10 @@ elif "linux2" == os.sys.platform or "linux3" == os.sys.platform:
     linux = True
     platform = "linux"
 
+    # use system-wide pcre
+    env.Append( CPPPATH=["/usr/include"] )
+    env.Append( LIBS=["pcrecpp", "pcre"] )
+
     if os.uname()[4] == "x86_64" and not force32:
         linux64 = True
         nixLibPrefix = "lib"
-- 
