From: Antonin Kral <A.Kral@sh.cvut.cz>
Date: Sat, 20 Apr 2013 22:05:50 -0300
Subject: Update SConscript.client to use "installAs" instead of "install".

---
 src/SConscript.client |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/SConscript.client b/src/SConscript.client
index 74bd73c..37e76a0 100644
--- a/src/SConscript.client
+++ b/src/SConscript.client
@@ -177,7 +177,7 @@ env.Install(
 # install
 prefix = GetOption("prefix")
 
-env.Install(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
+env.InstallAs(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
 
 for x in clientHeaderDirectories:
     inst = env.Install(prefix + "/include/mongo/" + x,
