--- a/SConstruct
+++ b/SConstruct
@@ -38,7 +38,9 @@
 # command-line. Treat this output as the value of HOST_OS
 def get_running_os_name():
     running_os = os.sys.platform
-    if running_os.startswith('linux') or running_os.startswith('gnukfreebsd'):
+    if (running_os.startswith('linux')
+        or running_os.startswith('gnukfreebsd')
+        or running_os.startswith('gnu0')):
         running_os = 'linux'
     elif running_os.startswith('freebsd'):
         running_os = 'freebsd'
--- a/src/mongo/platform/random.cpp
+++ b/src/mongo/platform/random.cpp
@@ -145,7 +145,7 @@
     return new WinSecureRandom();
 }
 
-#elif defined(__linux__) || defined(__sun) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#elif defined(__linux__) || defined(__sun) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
 
 class InputStreamSecureRandom : public SecureRandom {
 public:
