Description: fix GCC 5 FTBFS
 Add missing header for shell.
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Bug-Debian: https://bugs.debian.org/794931
Forwarded: not-needed
Last-Update: 2015-08-08

---

--- mongodb-2.4.14.orig/src/mongo/shell/linenoise_utf8.h
+++ mongodb-2.4.14/src/mongo/shell/linenoise_utf8.h
@@ -16,6 +16,7 @@
  */
 
 #include <boost/smart_ptr/scoped_array.hpp>
+#include <algorithm>
 #include <string.h>
 
 namespace linenoise_utf8 {
