--- a/src/mongo/util/logfile.cpp
+++ b/src/mongo/util/logfile.cpp
@@ -38,6 +38,7 @@
 #include "mongo/util/startup_test.h"
 #include "mongo/util/text.h"
 
+#define PAGESIZE sysconf(_SC_PAGESIZE)
 
 using namespace mongoutils;
 
@@ -48,16 +49,16 @@
             if( 0 && debug ) {
                 try {
                     LogFile f("logfile_test");
-                    void *p = malloc(16384);
+                    void *p = malloc(4*PAGESIZE);
                     char *buf = (char*) p;
-                    buf += 4095;
+                    buf += PAGESIZE - 1;
                     buf = (char*) (((size_t)buf)&(~0xfff));
                     memset(buf, 'z', 8192);
-                    buf[8190] = '\n';
-                    buf[8191] = 'B';
+                    buf[2 * PAGESIZE - 2] = '\n';
+                    buf[2 * PAGESIZE - 1] = 'B';
                     buf[0] = 'A';
-                    f.synchronousAppend(buf, 8192);
-                    f.synchronousAppend(buf, 8192);
+                    f.synchronousAppend(buf, 2 * PAGESIZE);
+                    f.synchronousAppend(buf, 2 * PAGESIZE);
                     free(p);
                 }
                 catch(DBException& e ) {
@@ -238,7 +239,11 @@
 
         fassert( 16144, charsToWrite >= 0 );
         fassert( 16142, _fd >= 0 );
+
+// Disabling alignment test on PPC64 due to a 64kB page size on PPC64 instead of 4kB on x86
+#ifndef __PPC64__
         fassert( 16143, reinterpret_cast<ssize_t>( buf ) % g_minOSPageSizeBytes == 0 );  // aligned
+#endif
 
 #ifdef POSIX_FADV_DONTNEED
         const off_t pos = lseek(_fd, 0, SEEK_CUR); // doesn't actually seek, just get current position
