Index: mongodb-2.6.11/SConstruct
===================================================================
--- mongodb-2.6.11.orig/SConstruct	2016-01-15 17:29:08.000000000 +0200
+++ mongodb-2.6.11/SConstruct	2016-01-15 17:37:54.000000000 +0200
@@ -474,7 +474,12 @@
 
                    PCRE_VERSION='8.37',
                    CONFIGUREDIR = '#' + scons_data_dir + '/sconf_temp',
-                   CONFIGURELOG = '#' + scons_data_dir + '/config.log'
+                   CONFIGURELOG = '#' + scons_data_dir + '/config.log',
+
+                   CFLAGS=os.environ.get("CFLAGS", "-O3").split(),
+                   CXXFLAGS=os.environ.get("CXXFLAGS", "-O3").split(),
+                   CPPFLAGS=os.environ.get("CPPFLAGS", "").split(),
+                   LINKFLAGS=os.environ.get("LDFLAGS", "").split(),
                    )
 
 if has_option("cache"):
